File: libasynDiscordWebHook-Nukkit/src/main/java/com/ialdrich23xx/discord/body/embed/components/Author.java
package com.ialdrich23xx.discord.body.embed.components;

import com.ialdrich23xx.discord.body.embed.base.IconURL;
import com.ialdrich23xx.discord.body.embed.base.Name;
import com.ialdrich23xx.discord.body.embed.base.Structure;
import com.ialdrich23xx.discord.body.embed.base.URL;

import java.util.Map;

public class Author extends Structure implements Name, URL, IconURL {

    public Author(String name) {
        this.setName(name);
    }

    public static Author make(String name) {
        return new Author(name);
    }

    @Override
    public Boolean build() {
        if (this.getName().isEmpty()) return false;

        if (!this.getUrl().isEmpty() && !this.urlBuild()) return false;
        if (!this.getIcon().isEmpty() && !this.iconBuild()) return false;

        return true;
    }

    @Override
    public Map<String, Object> toArray() {
        Map<String, Object> result = this.nameToArray();

        if (!this.getUrl().isEmpty()) result.putAll(this.urlToArray());
        if (!this.getIcon().isEmpty()) result.putAll(this.iconToArray());

        return result;
    }

    @Override
    public String toString() {
        return "Author(" + this.nameToString() + "," + this.urlToString() + "," + this.iconToString() + ")";
    }
}

File: libasynDiscordWebhook-PM/src/ialdrich23xx/libasynwebhook/discord/body/embed/components/Author.php
<?php

declare(strict_types=1);

namespace ialdrich23xx\libasynwebhook\discord\body\embed\components;

use ialdrich23xx\libasynwebhook\discord\body\embed\base\IconURL;
use ialdrich23xx\libasynwebhook\discord\body\embed\base\Name;
use ialdrich23xx\libasynwebhook\discord\body\embed\base\Structure;
use ialdrich23xx\libasynwebhook\discord\body\embed\base\URL;
use function array_merge;
use function strlen;

class Author extends Structure
{
    use URL;
    use IconURL;
    use Name;

    public function __construct(string $name) {
        $this->setName($name);
    }

    public static function make(string $name): self
    {
        return new self($name);
    }

    public function build(): bool
    {
        if (strlen($this->getName()) === 0) return false;

        if (strlen($this->getIcon()) !== 0 && !$this->iconBuild()) return false;
        if (strlen($this->getUrl()) !== 0 && !$this->urlBuild()) return false;

        return true;
    }

    public function toArray(): array
    {
        $result = $this->nameToArray();

        if (strlen($this->getIcon()) !== 0) $result = array_merge($result, $this->iconToArray());
        if (strlen($this->getUrl()) !== 0) $result = array_merge($result, $this->urlToArray());

        return $result;
    }

    public function toString(): string
    {
        return "Author(" . $this->nameToString() . "," . $this->urlToString() . "," . $this->iconToString() . ")";
    }
}